src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/PhpConfigReferenceDumpPass.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Definition\ArrayShapeGenerator;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\Loader\Configurator\AppReference;
+
+class PhpConfigReferenceDumpPass implements CompilerPassInterface
+{
+    private const REFERENCE_TEMPLATE = <<<'EOPHP'
+        <?php
+
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+        /**
+         * This class provides array-shapes for configuring the services and bundles of an application.
+         *
+         * Services declared with the config() method are autowired and autoconfigured by default.
+         *
+         * All this is auto-generated and is for apps only. Bundles SHOULD NOT rely on it.
+         *
+         * Example:
+         *
+         *     ```php
+         *     // config/services.php
+         *     namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+         *
+         *     return App::config([
+         *         'services' => [
+         *             'App\\' => [
+         *                 'resource' => '../src/',
+         *             ],
+         *         ],
+         *     ]);
+         *     ```
+         *
+         * @psalm-import-type ImportsConfig from AppReference
+         * @psalm-import-type ParametersConfig from AppReference
+         * @psalm-import-type ServicesConfig from AppReference
+         *{APP_TYPES}
+         */
+        final class App extends AppReference
+        {
+            {APP_SHAPE}
+            public static function config(array $config): array
+            {
+                return parent::config($config);
+            }
+        }
+
+        namespace Symfony\Component\Routing\Loader\Configurator;
```

Comment:
is this actually valid ? Don't we need to use the syntax with curly braces around the code when wanting to use several namespaces in the same PHP file ?

Reply:
Yes it's valid https://www.php.net/manual/en/language.namespaces.definitionmultiple.php, it works with or without curly braces